src/main/scala/com/wavesplatform/state2/StateWriter.scala

```diff
@@ -96,4 +114,19 @@ class StateWriterImpl(p: JavaMapStorage) extends StateReaderImpl(p) with StateWr
   }
 }
 
+object StateWriterImpl extends ScorexLogging {
+
+  private def withTime[R](f: => R): (R, Long) = {
+    val t0 = System.currentTimeMillis()
+    val r: R = f
+    val t1 = System.currentTimeMillis()
+    (r, t1 - t0)
+  }
+
+  def measurePersist[F[_] <: TraversableOnce[_], A](s: String)(fa: F[A])(f: F[A] => Unit): Unit = {
```

Comment:
Looks like the time has finally come to make use of [Metrics](http://metrics.dropwizard.io/3.2.2/)!